core/threaded/watchdog.c

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @author Benjamin Asch
+ * @author Edward A. Lee
+ * @copyright (c) 2023, The University of California at Berkeley.
+ * License: <a href="https://github.com/lf-lang/reactor-c/blob/main/LICENSE.md">BSD 2-clause</a>
+ * @brief Definitions for watchdogs.
+ */
+
+#include <assert.h>
+#include "watchdog.h"
+
+extern int _lf_watchdog_number;
+extern watchdog_t* _lf_watchdogs;
+
+/**
+ * @brief Initialize watchdog mutexes.
+ * For any reactor with one or more watchdogs, the self struct should have a non-NULL
+ * `reactor_mutex` field which points to an instance of `lf_mutex_t`.
+ * This function initializes those mutexes.
+ */
+void _lf_initialize_watchdog_mutexes() {
+    for (int i = 0; i < _lf_watchdog_number; i++) {
+        self_base_t* current_base = _lf_watchdogs[i].base;
+        if (current_base->reactor_mutex != NULL) {
+            lf_mutex_init((lf_mutex_t*)(current_base->reactor_mutex));
+        }
+    }
+}
+
+/**
```

Comment:
This comment can be removed. It already exists in the `.h` file.

Reply:
I don't see this comment (or function) in watchdog.h. (?)